GuruFX/FactoryBenchmark/Program.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using GuruFX.Core;
+using GuruFX.Core.Entities;
+
+namespace FactoryBenchmark
+{
+	internal class Program
+	{
+		private static readonly string[] RandomNameValues = { "Bob", "Jane", "Ed", "Rob", "Glen" };
+		private static int _x = 0;
+		private static string RandomName()
+		{
+			++_x;
+			_x = _x % RandomNameValues.Length;
+			return RandomNameValues[_x];
+		}
+
+		private const int MaxRandomValues = 1000000;
+		private static readonly List<int> RandomValues = new List<int>(MaxRandomValues);
+		private static int _r = 0;
+
+		private static int RandomValue()
+		{
+			++_r;
+			_r = _r % RandomValues.Count;
+			return RandomValues[_r];
+		}
+
+		private static void InitRandomValues()
+		{
+			Random r = new Random();
+			for (int j = 0; j < MaxRandomValues; j++)
+			{
+				RandomValues.Add(r.Next(MaxRandomValues));
+			}
+		}
+
+		public interface ITestObject
+		{
+			string Name { get; set; }
+		}
+
+		public class TestObject : ITestObject
+		{
+			public TestObject()
+			{
+			}
+
+			public TestObject(string name)
+			{
+				Name = name;
+			}
+
+			public string Name { get; set; } = string.Empty;
+		}
+
+		private static void Main(string[] args)
+		{
+			InitRandomValues();
+
+
+			/*
+				TestObjectFactory Results: 
+					Warmup Iterations: 1000
+					Iterations: 500000
+
+												 FactoryFunc : 00:00:00.0276911
+										FactoryFunc_WithArgs : 00:00:00.0301806
+								   CompiledExpression<T>(id) : 00:00:00.0309615
+								CompiledExpression(id, type) : 00:00:00.0348999
+											  CreateInstance : 00:00:00.0277758
+									  CreateInstanceWithArgs : 00:00:00.5938680
+				
+				EntityFactory Results:
+					Warmup Iterations: 1000
+					Iterations: 500000
+
+												 FactoryFunc : 00:00:00.6093253
+										FactoryFunc_WithArgs : 00:00:00.6281512
+								   CompiledExpression<T>(id) : 00:00:00.6336916
+								CompiledExpression(id, type) : 00:00:00.6302690
+											  CreateInstance : 00:00:00.6294698
+									  CreateInstanceWithArgs : 00:00:04.5770215
+			
+			 
+			 Creating Entities is sooo much slower... perhaps the auto property initializers add to this.
+			 Perhaps using ConcurrentDictionary instead of just Dictionary is also slowing things down?
+			 */
+
+			BenchmarkTestObjectFactory();
+
+			BenchmarkEntityFactory();
+		}
```

Comment:
Still need to finish up the ComponentFactory benchmarks. But too sleepy to do it tonight.